### ISSUE TITLE
Incorrect serialization/deserialization of the `over` clause

### ISSUE THREAD

**Issue by MCPN** [sqlalchemy/sqlalchemy#11422]
### Describe the bug

When serializing and deserializing a statement containing the `over` clause with `None` or `0` in range using `sqlalchemy.ext.serializer`, the resulting query contains a different range. Concrete ranges like `(1, 2)` seem to work fine.

The issue can be reproduced in both v2 and v1.4, the latter having a seemingly undefined parameters in the result.

While the serializer extension in considered legacy, we are (unfortunately) still using it in our RPC scenario with untrusted user inputs. Any tips on how to patch `over` on the user side or any alternatives to `sqlalchemy.ext.serializer` are much appreciated)
 

### Optional link from https://docs.sqlalchemy.org which documents the behavior that is expected

https://docs.sqlalchemy.org/en/20/core/serializer.html

### SQLAlchemy Version in Use

1.4.52, 2.0.30

### DBAPI (i.e. the database driver)

-

### Database Vendor and Major Version

-

### Python Version

3.12

### Operating system

Linux

### To Reproduce

```python
import sqlalchemy as sa
from sqlalchemy.ext.serializer import dumps as sa_dumps, loads as sa_loads


stmt = sa.func.SUM().over(range_=(None, 0))
dumped_query = sa_dumps(stmt)
loaded_query = sa_loads(dumped_query)
```

### Error

Expected result (the same as for the original statement):
```
>>> str(stmt.compile(compile_kwargs={"literal_binds": True}))
'sum() OVER (RANGE BETWEEN UNBOUNDED PRECEDING AND CURRENT ROW)'
```

Actual result for `SQLAlchemy-2.0.30`:
```
>> str(loaded_query.compile(compile_kwargs={"literal_binds": True}))
'sum() OVER (RANGE BETWEEN CURRENT ROW AND 1 FOLLOWING)'
``` 

Actual result for `SQLAlchemy-1.4.52`:
```
>>> str(loaded_query.compile(compile_kwargs={"literal_binds": True}))
'sum() OVER (RANGE BETWEEN 2152484616254935939 FOLLOWING AND 2545834031883131234 FOLLOWING)'
``` 
with exact numbers varying per run.

**Comment by CaselIT** [comment]
Hi,

It seems to be a bug in the Over implementation. Thanks for reporting

**Comment by sqla-tester** [comment]
**Federico Caselli** has proposed a fix for this issue in the **main** branch:

**Fix Over serialization** https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/5327

**Comment by sqla-tester** [comment]
**Federico Caselli** has proposed a fix for this issue in the **rel_2_0** branch:

**Fix Over serialization** https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/5328

**Comment by MCPN** [comment]
@CaselIT Thank you very much for the fix! Judging from milestones, it will be also included in the next 1.4.* release?

If so, are there any estimates on when the new release will be deployed? Only asking to understand if we should upgrade to 1.4.52 and apply the fix manually or we can wait a little and upgrade straight to 1.4.53?

**Comment by zzzeek** [comment]
this was milestoned to the 2.0 series only, not 1.4.   in 1.4 we typically only backport issues that would impact current working code, like connection pool race conditions and such.     the fix here is also local to the 2.0 series' architecture as it uses Enum - 1.4 still works on Python 2, so this would not be a clean backport.
